src/test/moment/create.js

```diff
@@ -750,6 +750,13 @@ test('parsing iso week year/week/weekday', function (assert) {
     assert.equal(moment.utc('2012-W01').format(), '2012-01-02T00:00:00Z', '2012 week 1 (1st Jan Sun)');
 });
 
+test('parsing weekday on utc dates verifies day acccording to local time', function (assert) {
+    assert.ok(moment.utc('Mon 03:59', 'ddd HH:mm').isValid(), 'Monday 03:59 (crosses UTC date');
```

Comment:
Er, it might cross the UTC date, depending on the time zone this was run in. I think this test would be better if it had the offset in the string.

Reply:
Yeah, I tried to emphasize this weakness in my description. Unfortunately adding an offset in the string won't make this test more universal, as the validity check fails because of a mismatch in the converted UTC day and user's local day. This bug would never be encountered when running moment somewhere with 0 UTC offset.

To illustrate:

```js
// Running on a computer with UTC offset -08:00
> moment.utc('Mon 06:59 -02:00)', 'ddd HH:mm Z')
moment.utc("2017-11-13T08:59:00.000+00:00")
> moment.utc('Mon 06:59 -01:00)', 'ddd HH:mm Z')
moment.invalid(/* Mon 06:59 -01:00) */)
> moment.utc('Mon 05:59 -03:00)', 'ddd HH:mm Z')
moment.utc("2017-11-13T08:59:00.000+00:00")
> moment.utc('Mon 05:59 -02:00)', 'ddd HH:mm Z')
moment.invalid(/* Mon 05:59 -02:00) */)

// Running on a computer with UTC offset -04:00
// Both previous failures pass:
> moment.utc('Mon 06:59 -01:00)', 'ddd HH:mm Z')
moment.utc("2017-11-13T07:59:00.000+00:00")
> moment.utc('Mon 05:59 -02:00)', 'ddd HH:mm Z')
moment.utc("2017-11-13T07:59:00.000+00:00")

// Running on a computer with UTC offset -04:00
// Fails on its own time of crossing date bounds with UTC
> moment.utc('Mon 03:59', 'ddd HH:mm')
moment.invalid(/* Mon 03:59 */)
> moment.utc('Mon 03:59 -01:00', 'ddd HH:mm Z')
moment.utc("2017-11-13T04:59:00.000+00:00")
> moment.utc('Mon 02:59 -01:00', 'ddd HH:mm Z')
moment.invalid(/* Mon 02:59 -01:00 */)
> moment.utc('Mon 02:59 -02:00', 'ddd HH:mm Z')
moment.utc("2017-11-13T04:59:00.000+00:00")

```

Let me know if you have any ideas for making a more robust test. I believe it would need to involve getting javascript's Date to use a mocked timezone for its day of week calculations.

Reply:
Ah, I see. Yeah, that's tricky.

Just thinking out loud, one trick would be to iterate across the hours and make sure they all work (i.e. that, at least for hour-aligned offsets, every time zone will get the right day). Of course, that still requires someone to actually run them in the relevant zones (e.g. presumably there's also a bug for the positive offset zones where it crosses in to the next day). So maybe that doesn't help much.

Another challenge is that the CI runs in UTC. I may write add running in America/New_York to the tests, in which case the test you've written is already ideal.

OK, let's merge this as-is.